modules/hal_silabs/simplicity_sdk/CMakeLists.txt

```
@@ -23,8 +23,16 @@ set(BLOBS_DIR      ${ZEPHYR_HAL_SILABS_MODULE_DIR}/zephyr/blobs/simplicity_sdk)
 # respectively.
 string(TOUPPER ${CONFIG_SOC_SERIES} SILABS_DEVICE_FAMILY)
 
-# Get SoC series number, i.e. translate e.g. efr32bg22 -> 22
-string(SUBSTRING ${CONFIG_SOC_SERIES} 7 2 SILABS_DEVICE_FAMILY_NUMBER)
+# Get SoC series number, i.e. translate efr32bg22 -> 22, mgm240p -> 24, simg301 -> 301
+if(CONFIG_SOC_SERIES MATCHES "^ef[rm]32")
+  string(SUBSTRING ${CONFIG_SOC_SERIES} 7 2 SILABS_DEVICE_FAMILY_NUMBER)
+elseif(CONFIG_SOC_SERIES MATCHES "^si")
+  string(SUBSTRING ${CONFIG_SOC_SERIES} 4 3 SILABS_DEVICE_FAMILY_NUMBER)
+elseif(CONFIG_SOC_SERIES MATCHES "^.gm")
+  string(SUBSTRING ${CONFIG_SOC_SERIES} 3 2 SILABS_DEVICE_FAMILY_NUMBER)
+else()
+  error("unknown part")
```

i don't think error exists in cmake?

```suggestion
  message("unknown part")
```